### PR TITLE
Update the `oauth/*` endpoints

### PR DESCRIPTION
**Pull request by shannynalayna**
Updates the `oauth/login` endpoint to add the `redirect_url` query
parameter. Also updates the `oauth/callback` endpoint to update the
response to the redirect and explain the contents of that
redirect URL.

Signed-off-by: Shannyn Telander <telander@bitwise.io>

### CODE REVIEW

File: splinterd/api/static/openapi.yaml [hunk 1]
```diff
@@ -1577,6 +1577,14 @@ paths:
         authenticate the user.
       parameters:
         - $ref: "#/components/parameters/protocol_version"
+        - name: redirect_url
+          in: query
+          description: |-
```
Comment (ltseeley) [inline splinterd/api/static/openapi.yaml:1582]: Is the `-` here intentional?

File: splinterd/api/static/openapi.yaml [hunk 2]
```diff
@@ -1577,6 +1577,14 @@ paths:
...
+            The URL the client will be redirected to once the OAuth
+            authentication process has been completed.
```
Comment (ltseeley) [inline splinterd/api/static/openapi.yaml:1584]: Should explain what happens when the parameter isn't provided.

File: splinterd/api/static/openapi.yaml [hunk 3]
```diff
@@ -1603,28 +1613,26 @@ paths:
       tags:
         - OAuth
       description: |
-        Receives an authorization code from the OAuth provider and exchanges it
-        for an access token for the user.
+        Receives an authorization code from the OAuth provider to create an
+        access token for the user. Redirects to a URL provided by the client
+        when the OAuth authentication process is initiated.
```
Comment (ltseeley) [inline splinterd/api/static/openapi.yaml:1618]: s/is/was/

File: splinterd/api/static/openapi.yaml [hunk 4]
```diff
@@ -1603,28 +1613,26 @@ paths:
...
       parameters:
         - $ref: "#/components/parameters/protocol_version"
       responses:
-        200:
-          description: Obtained access token for user
-          content:
-            application/json:
+        302:
+          description: Found redirect
```
Comment (ltseeley) [inline splinterd/api/static/openapi.yaml:1623]: This description should explain what this redirect means

**Review by ltseeley — CHANGES_REQUESTED** [review]

File: splinterd/api/static/openapi.yaml (hunk 1 above)
Comment (shannynalayna) [reply splinterd/api/static/openapi.yaml:1582]: `|-` seems to be used interchangeably with just `|` throughout the doc, I was just following an example that used `-`

File: splinterd/api/static/openapi.yaml (hunk 2 above)
Comment (shannynalayna) [reply splinterd/api/static/openapi.yaml:1584]: I specified this in the 400 response description but can add a small explanation here as well. Was kind of awkward finding just the right place to put these descriptions...

**Review by ltseeley — APPROVED** [review]

**Review by peterschwarz — APPROVED** [review]

**Pull request merged**